src/Symfony/Component/DependencyInjection/Compiler/ResolveReferencesToAliasesPass.php

```diff
@@ -66,7 +66,7 @@ private function getDefinitionId($id, ContainerBuilder $container)
         $seen = array();
         while ($container->hasAlias($id)) {
             if (isset($seen[$id])) {
-                throw new ServiceCircularReferenceException($id, array_keys($seen));
+                throw new ServiceCircularReferenceException($id, array_merge(array_keys($seen), array($id)));
```

Comment:
array_unique(array_merge($a, $b)) -> I think this will remove duplicates in case of the merge